service/store/store.go

```diff
@@ -517,30 +520,20 @@ func (s *Store) dealDataFilePathFor(bidID auction.BidID, payloadCid string) stri
 }
 
 // enqueueDataURI queues a data uri fetch.
-// commitTxn will be committed internally!
-func (s *Store) enqueueDataURI(commitTxn ds.Txn, b *Bid) error {
-	// Set the bid to "fetching_data"
-	if err := s.saveAndTransitionStatus(commitTxn, b, BidStatusFetchingData); err != nil {
-		return fmt.Errorf("updating status (fetching_data): %v", err)
-	}
-	if commitTxn != nil {
-		if err := commitTxn.Commit(); err != nil {
-			return fmt.Errorf("committing txn: %v", err)
+func (s *Store) enqueueDataURI(txn ds.Txn, b *Bid) error {
+	select {
+	case s.jobCh <- b:
+		// Set the bid to "fetching_data"
+		if err := s.saveAndTransitionStatus(txn, b, BidStatusFetchingData); err != nil {
+			return fmt.Errorf("updating status (fetching_data): %v", err)
 		}
-	}
-
-	// Unblock the caller by letting the rest happen in the background
-	go func() {
-		select {
-		case s.jobCh <- b:
-		default:
-			log.Debugf("workers are busy; queueing %s", b.ID)
-			// Workers are busy, set back to "queued_data"
-			if err := s.saveAndTransitionStatus(nil, b, BidStatusQueuedData); err != nil {
-				log.Errorf("updating status (queued_data): %v", err)
-			}
+	default:
+		log.Debugf("workers are busy; queueing %s", b.ID)
+		// Workers are busy, set back to "queued_data"
+		if err := s.saveAndTransitionStatus(txn, b, BidStatusQueuedData); err != nil {
```

Comment:
wrap in transaction too.

Reply:
well. this makes no sense. we should try best to set back to `BidStatusQueuedData`. reverted as well.

Reply:
Okay so reverted again, since we always want this switching to `BidStatusQueuedData` happens after the switching to `BidStatusFetchingData` above at L525, and if it's not wrapped in the same transaction, the order is reversed - the transaction created inside `saveAndTransitionStatus` be committed first.

Reply:
LGTM!